health/notifications/alarm-notify.sh.in

```diff
@@ -452,6 +452,7 @@ filter_recipient_by_criticality() {
 		if [ -f "${NETDATA_CACHE_DIR}/alarm-notify/${method}/${r}/${alarm_id}" ]; then
 			# we do not remove the file, so that he will get future notifications of this alarm
 			debug "SEVERITY FILTERING for ${x} VIA ${method}: ALLOW: recipient has been notified for this alarm in the past (will still receive next status change)"
+			return 0
```

Comment:
This line is returning for the previous state without fix the bug that user had requested and it was "fixed" removing these lines.. I configured my `health_alarm_notify.conf` as:

```
DEFAULT_RECIPIENT_SLACK="channel|critical"
```

As result I am receiving all notifications on slack:

![err1](https://user-images.githubusercontent.com/49162938/89790240-2cc11600-db11-11ea-815b-e37f2f2072c6.png)

and on Telegram:

![err](https://user-images.githubusercontent.com/49162938/89790367-5c701e00-db11-11ea-88bc-bea636046372.png)

On Telegram this is expected, but on Slack I could not receive the `warn`.

Reply:
The WARNING can't be sent after the CLEAR. CLEAR removes the file that is touched by CRITICAL. WARNING can be sent in only two conditions: 
- Another rule that allows WARNING to be sent
- WARNING issued before CLEAR has had a chance to remove the file.

The timestamps in these examples are too close together to tell if you're in the race condition, but running alarm-notify.sh in debug mode will prove that it was working as designed, in the first place.

- Run `bash -x /usr/libexec/netdata/plugins.d/alarm-notify.sh test`, with debug enabled (https://learn.netdata.cloud/docs/agent/health/notifications#testing-notifications)
- Look in the output for the alarm-notify.sh calls. There are three of them. 

Run them one by one and after each execution look at the debug output and also check for the existence of /var/cache/netdata/alarm-notify/xxx/zzz

You'll see that CRITICAL creates a file there, CLEAR deletes it. Depending on the existence of that file, other notifications are sent (including ones for WARNING).

Reply:
@cakrit I agree with everything you wrote, and I can confirm that when we use the shell script passing `test` as argument, we have it working as we expect. The problem is happening when we use an alarm, for example, I used this alarm:

```
 alarm: dev_dim_template
    on: system.cpu
    os: linux
lookup: sum -3s at 0 every 3 percentage foreach *
 units: %
 every: 10s
  warn: $this > 1
  crit: $this > 4
```

This is not a real alarm, but at least it has a high frequency that helps me to debug the feature. With this alarm, you will see that the channels are receiving `warn` alarms instead only critical and clean:

![err3](https://user-images.githubusercontent.com/49162938/89817462-aff56280-db37-11ea-8cfe-8eddebbab8fc.png)

Reply:
You are supposed to receive all alarms, AFTER the first condition fires. 
So if you have WARNING, WARNING, CRITICAL, WARNING, CLEAR, you should receive 3 notifications (CRITICAL, WARNING CLEAR).  
If you have CRITICAL WARNING CRITICAL WARNING CLEAR, you should receive 5 notifications.

2 warnings one after the other doesn't make sense BTW.

Reply:
I won't disagree of you here, but I think you are missing something, and I will argue with some links the motive we either need to remove this `return 0` from here or we need to be more explicit inside our [health_alarm_notify.conf file](https://github.com/netdata/netdata/blob/master/health/notifications/health_alarm_notify.conf#L166).

According the `health_alarm_notify.conf` if the users specify `|critical` they should not receive warning in that channel, so I think there was not an erroneous [bug report](https://github.com/netdata/netdata/issues/8739#issuecomment-671246311).

My final request to approve this PR is, please, can you either update `health_alarm_notify.conf` comments or to remove this `return 0`?

I agree that the previous PR made a generalization to also remove `clear` notification and I continue thinking that we in a near future should bring more possibilities for the user specify all possible alarms using pipes.

Reply:
I put that in https://github.com/netdata/netdata/pull/9740.
Laptop issues, can't add commit to this PR. Merge both of them on the same day